addons/sale_mrp/sale_mrp.py

```diff
@@ -41,9 +41,11 @@ def _get_delivered_qty(self):
         for bom in self.product_id.product_tmpl_id.bom_ids:
             if bom.type != 'phantom':
                 continue
+            ctx = dict(self.env.context or {})
+            ctx.update({'bom_effectivity_date': getattr(self.order_id, 'effective_date', self.order_id.date_order)})
```

Comment:
The comment near the beginning of this method says "we don't know which [BoM] was used when the delivery was created." and yet when it calls `_bom_explode` and ultimately `_skip_bom_line` and `_bom_find` it will choose a BOM with an effective date of `time.strftime(DEFAULT_SERVER_DATE_FORMAT)` ([L218](https://github.com/odoo/odoo/pull/11373/files#diff-200a0bd9e95f58f31b37039e65f28d69L218), [L240](https://github.com/odoo/odoo/pull/11373/files#diff-200a0bd9e95f58f31b37039e65f28d69L240)) not the "Date on which the first Delivery Order was created" (tooltip for SO Effective Date) which is what this addition does.